.github/workflows/publish-preview.yml

```diff
@@ -0,0 +1,34 @@
+name: Publish Preview to NuGet
+
+on:
+  push:
+    tags:
+    - "v[0-9]+.[0-9]+.[0-9]+-preview-[0-9]+"
+    - "v[0-9]+.[0-9]+.[0-9]+-alpha-[0-9]+"
+    - "v[0-9]+.[0-9]+.[0-9]+-beta-[0-9]+"
+    - "v[0-9]+.[0-9]+.[0-9]+-prerelease-[0-9]+"
+    - "v[0-9]+.[0-9]+.[0-9]+-rc-[0-9]+"
+
+jobs:
+  preview:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 6.0.x
+    - name: Restore Dependencies
+      run: cd src && dotnet restore
+    - name: Build
+      run: cd src && dotnet build --configuration Release --no-restore
+    - name: Set VERSION variable from tag
+      run: echo "VERSION=${GITHUB_REF/refs\/tags\/v/}" >> $GITHUB_ENV
+    - name: Pack NuGet
+      run: cd src && dotnet pack --no-build --configuration Release /p:Version=${VERSION} --output .
+    - name: Publish NuGet
+      run: cd src && dotnet nuget push *.nupkg --source https://api.nuget.org/v3/index.json --api-key ${NUGET_KEY} --skip-duplicate

```

Comment:
Would it make more sense to provide a path to the solution file rather than using `cd`?

Reply:
Let me give that a try. I think another idea would be to set the working directory to `src` once, if we can do that in GitHub Actions.